demos/custom-tabs-headers/src/main/java/com/google/androidbrowserhelper/demos/customtabsheaders/MainActivity.java

```diff
@@ -0,0 +1,120 @@
+/*
+ *    Copyright 2020 Google LLC
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *    https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+package com.google.androidbrowserhelper.demos.customtabsheaders;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.appcompat.app.AppCompatActivity;
+import androidx.browser.customtabs.CustomTabsCallback;
+import androidx.browser.customtabs.CustomTabsClient;
+import androidx.browser.customtabs.CustomTabsIntent;
+import androidx.browser.customtabs.CustomTabsService;
+import androidx.browser.customtabs.CustomTabsServiceConnection;
+import androidx.browser.customtabs.CustomTabsSession;
+
+import android.content.ComponentName;
+import android.content.Context;
+import android.net.Uri;
+import android.os.Bundle;
+import android.provider.Browser;
+import android.widget.Button;
+import android.widget.Toast;
+
+public class MainActivity extends AppCompatActivity {
+    private static final Uri URL = Uri.parse("https://padr31.github.io/");
+
+    private CustomTabsSession mSession;
+    private CustomTabsServiceConnection mConnection;
+
+    @Override
+    protected void onStart() {
+        super.onStart();
+
+        // Set up a callback that launches the intent after session was validated.
+        CustomTabsCallback callback = new CustomTabsCallback() {
+            @Override
+            public void onRelationshipValidationResult(int relation, @NonNull Uri requestedOrigin,
+                    boolean result, @Nullable Bundle extras) {
+                // Can launch custom tabs intent after session was validated as the same origin.
+                findViewById(R.id.btn_extra).setEnabled(true);
```

Comment:
nit: the code is calling `findViewById(R.id.btn_extra)` a few times. To void this, create a `private Button mExtraButton` as an instance variable and set `mExtraButton = findViewById(R.id.extra)` on `onCreate()`. Then, just use the instance variable across the code.

This pattern is quite common on Android. Besides using the "Don't Repeat Yourself" (DRY) concept, `findViewById()` is also O(n). Even though in practice this shouldn't be an issue for a simple layout.